src/amo/components/FeedbackForm/index.js

```diff
@@ -0,0 +1,439 @@
+        reason: category,
+      }),
+    );
+  };
+
+  getFormValues(currentUser: UserType | null): FormValues {
+    const defaultFormValues = {
+      email: '',
+      name: '',
+      text: '',
+      category: null,
+      legalAssertion: false,
+    };
+
+    const { email, display_name: name } = currentUser || {};
+
+    return {
+      ...defaultFormValues,
+      email: email || '',
+      name: name || '',
+    };
+  }
+
+  preventSubmit(): boolean {
+    const { loading } = this.props;
+    const { text, category, legalAssertion } = this.state;
+
+    return (
+      loading ||
+      !category ||
+      !text ||
+      (text && text.trim() === '') ||
+      !legalAssertion
+    );
+  }
+
+  renderReportSentConfirmation(i18n: I18nType): React.Node {
+    return (
+      <div>
+        <Card
+          header={i18n.gettext('You sent a report about this add-on')}
+          className="FeedbackForm--Card"
+        >
+          <p className="ReportAbuseButton-first-paragraph">
+            {i18n.gettext(
+              `We have received your report. Thanks for letting us know.`,
+            )}
+          </p>
+        </Card>
+      </div>
+    );
+  }
+
+  render(): React.Node {
+    const { addonId, abuseReport, currentUser, errorHandler, i18n, loading } =
+      this.props;
+
+    let errorMessage;
+    if (
+      errorHandler.hasError() &&
+      ![401, 403, 404].includes(errorHandler.capturedError.responseStatusCode)
+    ) {
+      errorMessage = errorHandler.renderError();
+    }
+
+    const submitButtonText = loading
+      ? i18n.gettext('Submitting your report…')
+      : i18n.gettext('Submit report');
+
+    const categories = getCategories(i18n);
+    const categoryInputs = {};
+    // eslint-disable-next-line guard-for-in
+    for (const categoryType in categories) {
+      categoryInputs[categoryType] = [];
+      categories[categoryType].forEach((category) => {
+        categoryInputs[categoryType].push(
+          <li className="FeedbackForm-category--li" id={`li-${category.value}`}>
+            <input
+              type="radio"
+              className="FeedbackForm-catgeory"
+              id={`feedbackCategory${category.value}`}
+              name="category"
+              onChange={this.onFieldChange}
+              value={category.value}
+              selected={this.state.category === category.value}
+            />
+            <label
+              className="FeedbackForm--label"
+              htmlFor={`feedbackCategory${category.value}`}
+            >
+              {category.label}
+            </label>
+            <p className="FeedbackForm--help">{category.help}</p>
+          </li>,
+        );
+      });
+    }
+
+    const abuseSubmitted = abuseReport && abuseReport.message !== undefined;
+
+    return (
+      <div className="FeedbackForm">
+        <Helmet>
+          <title>
+            {i18n.sprintf(
+              i18n.gettext('Submit FeedbackForm for %(type)s %(identifier)s'),
```

Comment:
```suggestion
              i18n.gettext('Submit feedback or report for %(type)s %(identifier)s'),
```

Something like that ? `FeedbackForm` shouldn't be in the string.

Reply:
good spot - it was originally just "Feedback" but I guess it got caught up in Will's find&replace when he refactored most of the code from a `Feedback` page to a `FeedbackForm` component.